### DISCUSSION TITLE
Problem when building slides 2_2-ownership-and-references

### DISCUSSION THREAD

**Discussion by goto40** [trifectatechfoundation/teach-rs#139] [Q&A] [answered]
Thank your for this great resource you provided! It looks really great.

I just played around with it and found a small problem (probably due to some wrong usage or wrong dependencies from my side):

When building slides from the generated full track (`cargo run generate -o ~/temp/teach-rs-full ../content/full.track.toml`) I get the following:

```
npm i
npm run build-2_2
...
✓ 254 modules transformed.
x Build failed in 1.70s
[vite:vue] [plugin vite:vue] 2_2-ownership-and-references.md__slidev_22.md (32:1): Invalid end tag.
```

While exporting I get some more explicit errors:
```
npm run export-2_2
...
  ◆ rendering...10:26:58 AM [vite] Pre-transform error: Invalid end tag.
10:26:58 AM [vite] Pre-transform error: Invalid end tag.
  ▲ rendering...10:26:58 AM [vite] Internal server error: Invalid end tag.
  Plugin: vite:vue
  File: /home/pierre/temp/teach-rs-full/slides/2_2-ownership-and-references.md__slidev_22.md:1
  30 |  </ul>
  ✓ exported to ./2_2-ownership-and-references-export.pdf
  32 |  </CodeBlockWrapper>
...
  33 |  </v-click>
  34 |  </InjectedLayout>
```

Any hint how to debug this is appreciated...
Is this some wrong usage or a bug?

**Comment by goto40** [comment]
I found a way to "fix" the problem:

* The PDF-exported-slides indicated the problematic slide
* Playing around I found that the text showing an indented compiler output was causing the error: removing the indentation fixed the problem. However, the out put has now line numbers (which I somehow did not expect).
* I do not understand why this happened - so I do not yet create a PR

![image](https://github.com/user-attachments/assets/35b90706-82ef-4e9e-9091-affba79e104b)

![image](https://github.com/user-attachments/assets/146372d4-7456-4a47-88bb-ec45436f0429)

**Comment by goto40** [comment]
Maybe this is related (thy have a similar error - but I could not yet figure out how to use the info): https://github.com/slidevjs/slidev/issues/1865

**Comment by goto40** [comment]
I further digged into slidev...

* "@slidev/cli": "0.49.29" --> problem happens
* "@slidev/cli": "0.49.20" --> problem happens
* "@slidev/cli": "0.49.10" --> no problem :-)
* "@slidev/cli": "0.50.0" --> also no problem :-)

It seems to be a temporary problem with slidev/cli between 0.49.20 and 0.49.29...

**Comment by hdoordt** [comment, marked as answer]
Thanks for figuring this out! [This commit](https://github.com/trifectatechfoundation/teach-rs/commit/b62c6c147cc74a2b77301574e0fc846e66939894) updates slidev to v51, and should therefore fix your problem.
